pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -88,13 +88,22 @@ func Add(mgr manager.Manager, context *controllerconfig.Context) error {
 	}
 
 	pdbPredicate := predicate.Funcs{
```

Comment:
Remind me why we need to watch the PDBs? Doesn't the operator act on other events already?

Reply:
After removing the node drain canary,  we needed something that would tell that operator that OSD is down ( or drained) and it should reconcile. This was not possible with the events (other than PDB) that the reconcile acts upon. 

-  Earlier, we were dependent on the cephCluster health check status updating every 60 seconds.  This was not intentional but it did the job. But it was not a good way as user can change the healthCheck duration.  So we decided to make the operator reconcile more proactive. 
- One suggested option was to reconcile every 30 seconds irrespective of any changes. 
- Another approach (currently implemented) was to watch for the Main OSD PDB. With one OSD down (or drained) , the AllowedDisruption of this Main PDB will be 0.  This is what we use to trigger the reconcile now. More in the PR description [here](https://github.com/rook/rook/pull/6749)

Reply:
Thanks for the reminder. How about adding a comment here next to the predicate for future reference so I don't ask again? :)

Reply:
added comments.